src/engraving/libmscore/scorefile.cpp

```diff
@@ -82,6 +82,10 @@ void Score::write(XmlWriter& xml, bool selectionOnly, compat::WriteScoreHook& ho
     // then some layout information is missing:
     // relayout with all parts set visible
 
+    if (selectionOnly && _selection.staffStart() == _selection.staffEnd()) {
+        return;
```

Comment:
Maybe better check this condition before opening the file (NotationProject::saveSelectionOnScore or NotationProject::writeProject), so that an empty file is not created.
We can also return a "Nothing selected" error so that we can show the result to the user.